File: rag-doc-search/frontend/my_component.js
function downloadFile(file) {
    // ファイルをBlobに変換
    const blob = new Blob([file], { type: file.type });
  
    // Python側の関数を呼び出す
    // 'pythonFunctionName' はPython側の関数の名前です。
    // 'fileData' はPython側に渡すファイルデータです。
    streamlit.components.v1.html("pythonFunctionName", { fileData: blob }); 
  }
  
  async function fetchData() {
    try {
      const response = await fetch("https://example.com/api/file"); // APIエンドポイントを指定
      if (!response.ok) {
        throw new Error(`API request failed with status ${response.status}`);
      }
      const fileBlob = await response.blob();
      downloadFile(fileBlob);
    } catch (error) {
      console.error("Error fetching data:", error);
    }
  }
  
  fetchData();

File: rag-doc-search/frontend/my_component.py
import streamlit as st
from streamlit.components.v1 import html
from io import BytesIO
import base64

def pythonFunctionName(fileData):
  # ファイルデータをPDFに変換
  pdf_bytes = fileData.read() # バイナリデータとして読み込み
  pdf_buffer = BytesIO(pdf_bytes)

  # PDFをStreamlitに表示
  base64_pdf = base64.b64encode(pdf_buffer.getvalue()).decode('utf-8')
  pdf_display = st.markdown(f"""
  <iframe src="data:application/pdf;base64,{base64_pdf}" width="100%" height="600px">
  </iframe>
  """, unsafe_allow_html=True)

html("my_component.js")